nixos/modules/services/networking/bird.nix

```diff
@@ -68,36 +69,38 @@ in
   };
 
   imports = [
-    (lib.mkRemovedOptionModule [ "services" "bird" ] "Use services.bird2 instead")
-    (lib.mkRemovedOptionModule [ "services" "bird6" ] "Use services.bird2 instead")
+    (lib.mkRemovedOptionModule [ "services" "bird2" ]
+      "Use services.bird instead. bird3 is the new default bird package. You can choose to remain with bird2 by setting the service.bird.package option."
+    )
+    (lib.mkRemovedOptionModule [ "services" "bird6" ] "Use services.bird instead")
   ];
 
   ###### implementation
   config = mkIf cfg.enable {
-    environment.systemPackages = [ pkgs.bird ];
+    environment.systemPackages = [ cfg.package ];
 
-    environment.etc."bird/bird2.conf".source = pkgs.writeTextFile {
-      name = "bird2";
+    environment.etc."bird/bird.conf".source = pkgs.writeTextFile {
+      name = "bird";
       text = cfg.config;
       checkPhase = optionalString cfg.checkConfig ''
-        ln -s $out bird2.conf
+        ln -s $out bird.conf
         ${cfg.preCheckConfig}
-        ${pkgs.buildPackages.bird}/bin/bird -d -p -c bird2.conf
+        ${pkgs.buildPackages.bird}/bin/bird -d -p -c bird.conf
       '';
     };
```

Comment:
I think the package option should be used here. I don't know hot this should be handled in regard to cross compilation.

I've tested the behavior in a repl and this should work.
```nix
    environment.etc."bird/bird.conf".source = pkgs.writeTextFile {
      name = "bird";
      text = cfg.config;
      derivationArgs.nativeBuildInputs = [ cfg.package ];
      checkPhase = optionalString cfg.checkConfig ''
        ln -s $out bird.conf
        ${cfg.preCheckConfig}
        bird -d -p -c bird.conf
      '';
    };
```

Reply:
Good point, changed that.